src/plugins/discover/public/application/main/utils/resolve_data_view.ts

```diff
@@ -104,9 +101,10 @@ export async function loadDataView(
     fetchId = dataViewSpec.id!;
   }
 
+  let fetchedDataView: DataView | undefined;
   // try to fetch adhoc data view first
   try {
-    const fetchedDataView = fetchId ? await dataViews.get(fetchId) : undefined;
+    fetchedDataView = fetchId ? await dataViews.get(fetchId) : undefined;
```

Comment:
Previously this code was loading a data view, checking if it was adhoc, then tossing the result. Now we hold on to it in case its the data view we wanted to load.